python/doc/examples/meta_modeling/polynomial_chaos_metamodel/plot_enumeratefunction.py

```diff
@@ -0,0 +1,154 @@
+    if td > td_prev:
+        td_prev = td
+        print("---+-------------+-------------")
+    print(f"{i:2} |       {multi_index} |           {td}")
+
+# %%
+# Plot the multi-indices of the linear enumeration rule by stratas.
+# In the specific case of the linear enumerate function each strata contains
+# multi-indices of identical total degree that is also the index of the strata.
+def draw_stratas(enum_func):
+    maximum_strata_index = 7
+    graph = ot.Graph("", "$\\alpha_1$", "$\\alpha_2$", True)
+    if enum_func.__class__.__name__ == "LinearEnumerateFunction":
+        graph.setTitle("Linear enumeration rule")
+    elif enum_func.__class__.__name__ == "HyperbolicAnisotropicEnumerateFunction":
+        graph.setTitle(f"q={enum_func.getQ()}")
+    offset = 0
+    for strata_index in range(maximum_strata_index):
+        strata_cardinal = enum_func.getStrataCardinal(strata_index)
+        sample_in_layer = [enum_func(idx + offset) for idx in range(strata_cardinal)]
+        offset += strata_cardinal
+        cloud = ot.Cloud(sample_in_layer)
+        cloud.setLegend(str(strata_index))
+        cloud.setPointStyle("circle")
+        graph.add(cloud)
+    palette = ot.DrawableImplementation.BuildDefaultPalette(maximum_strata_index)
+    graph.setColors(palette)
+    graph.setIntegerXTick(True)
+    graph.setIntegerYTick(True)
+    graph.setLegendPosition("topright")
+    return graph
+graph = draw_stratas(enum_func)
+legend_kw={"bbox_to_anchor": (1.0, 1.0), "loc": "upper left"}
+view = otv.View(graph, legend_kw=legend_kw, axes_kw={"aspect": "equal"})
+
+# %%
+# Plot the number of terms in the basis depending on the maximum total degree
+# for several dimension values.
+# We observe the exponential increase of the number of terms with the dimension
+# :math:`d` (curse of dimensionality).
+graph = ot.Graph("Full tensorized basis", "Total degree", "Number of coefficients", True)
+degree_maximum = 10
+list_of_polynomial_degrees = [1, 5, 10, 15, 20]
+point_styles = ["bullet", "circle", "fdiamond", "fsquare", "triangleup"]
+palette = ot.DrawableImplementation.BuildDefaultPalette(len(list_of_polynomial_degrees))
+for i in range(len(list_of_polynomial_degrees)):
+    d = list_of_polynomial_degrees[i]
+    number_of_coeff_array = ot.Sample(degree_maximum, 2)
+    for p in range(1, 1 + degree_maximum):
+        number_of_coeff_array[p - 1, 0] = p
+        number_of_coeff_array[p - 1, 1] = m.comb(p + d, p)
+    cloud = ot.Cloud(number_of_coeff_array)
+    cloud.setPointStyle(point_styles[i])
+    cloud.setLegend(f"$d={d}$")
+    cloud.setColor(palette[i])
+    graph.add(cloud)
+graph.setLegendPosition("topleft")
+graph.setIntegerXTick(True)
+graph.setLogScale(ot.GraphImplementation.LOGY)
+view = otv.View(graph, figure_kw={"figsize": (5, 4)})
+
+# %%
+# Plot the hyperbolic quasi norm for different values of q.
+# With q=1 stratas are hyperplanes, and in case of isotropy
+# it is equivalent to the linear enumeration rule.
+def draw_qnorm(q):
```

Comment:
Defining a quasi-norm class instead of a lambda-function clarifies the script.

```suggestion
class QuasiNorm(ot.OpenTURNSPythonFunction):
    def __init__(self, dimension, quasi_norm_parameter):
        super(QuasiNorm, self).__init__(dimension, 1)
        self.quasi_norm_parameter = quasi_norm_parameter

    def _exec(self, x):
        x = ot.Point(x)
        dimension = x.getDimension()
        norm = 0.0
        for i in range(dimension):
            norm += x[i] ** self.quasi_norm_parameter
        norm = np.exp(np.log(norm) / self.quasi_norm_parameter)
        return [norm]

def draw_qnorm(q):
```